### PR TITLE
SITL: move sprayer and grippers in Aircraft, disable them by default

### PR DESCRIPTION
**Pull request by khancyr**
This prevent sprayer and gripper to be active when using octo on SITL. In more general way, disable sprayer and grippers by default
This allow to use those on other vehicles too
Servo assigned are now parametrable

### CODE REVIEW

**Comment by khancyr** [comment]
revert headers to C version

File: libraries/SITL/SIM_Aircraft.h [hunk 1]
```diff
@@ -31,6 +35,7 @@ namespace SITL {
  */
 class Aircraft {
     friend class Gripper_Servo;
+    friend class Sprayer;
```
Comment (khancyr) [inline libraries/SITL/SIM_Aircraft.h]: this is useless ...

**Comment by khancyr** [comment]
@peterbarker it should be better that way !

**Comment by khancyr** [comment]
need param handling rework

**Comment by khancyr** [comment]
@peterbarker the use of AP_SUBGROUPINFO in AP_SUBGROUPEXTENSION doesn't work ... Here for example, we need to  set SPR_ENABLE = 1 to have access to all other param like GRPS and GRPE subgroup ... 
Any idea of why it failed ?

File: libraries/SITL/SIM_Sprayer.h [hunk 2]
```diff
@@ -18,36 +18,39 @@
...
-private:
+ private:
 
     const uint32_t report_interval = 1000000; // microseconds
     uint64_t last_report_us;
 
-    const float pump_max_rate = 0.01; // litres/second
-    const float pump_slew_rate = 20; // percent/scond
+    const float pump_max_rate = 0.01f; // litres/second
+    const float pump_slew_rate = 20.0f; // percent/scond
```
Comment (amilcarlucas) [inline libraries/SITL/SIM_Sprayer.h]: s/scond/second/

**Comment by khancyr** [comment]
rebased and fix path for parameters

File: libraries/SITL/SIM_Aircraft.cpp [hunk 3]
```diff
@@ -737,4 +736,27 @@ void Aircraft::extrapolate_sensors(float delta_time)
...
+    // update gripper
+    if (sitl->gripper_sim.gripper_enable) {
+        const int16_t gripper_pwm= sitl->gripper_sim.gripper_servo_pin >= 0 ? input.servos[sitl->gripper_sim.gripper_servo_pin] : -1;
+        sitl->gripper_sim.update(gripper_pwm, on_ground());
+        external_payload_mass += sitl->gripper_sim.payload_mass(hagl());
+    }
+    if (sitl->gripper_epm_sim.gripper_emp_enable) {
+        const int16_t gripper_pwm = sitl->gripper_epm_sim.gripper_emp_servo_pin >= 0 ? input.servos[sitl->gripper_epm_sim.gripper_emp_servo_pin] : -1;
+        sitl->gripper_epm_sim.update(gripper_pwm);
+        // external_payload_mass += gripper_epm.payload_mass();
```
Comment (peterbarker) [inline libraries/SITL/SIM_Aircraft.cpp]: Why is this commented out?

File: libraries/SITL/SIM_Gripper_EPM.cpp [hunk 4]
```diff
@@ -83,7 +104,7 @@ bool Gripper_EPM::should_report()
         return false;
     }
 
-    if (fabs(reported_field_strength - field_strength) > 10.0f) {
```
Comment (peterbarker) [inline libraries/SITL/SIM_Gripper_EPM.cpp]: These are floats - why the change?

File: libraries/SITL/SIM_Gripper_EPM.cpp [hunk 5]
```diff
@@ -94,6 +115,6 @@ float Gripper_EPM::tesla()
 {
     // https://en.wikipedia.org/wiki/Orders_of_magnitude_(magnetic_field)
     // 200N lifting capacity ~= 2.5T
-    const float percentage_to_tesla = 0.25;
-    return percentage_to_tesla * field_strength/100.0f;
+    const float percentage_to_tesla = 0.25f;
+    return static_cast<float>(percentage_to_tesla * field_strength/100.0f);
```
Comment (peterbarker) [inline libraries/SITL/SIM_Gripper_EPM.cpp]: Some people aren't really keen on these static-casts.  Adding them into SITL-only code probably isn't the best of ideas.

File: libraries/SITL/SIM_Aircraft.cpp [hunk 6]
```diff
@@ -737,4 +736,27 @@ void Aircraft::extrapolate_sensors(float delta_time)
     velocity_air_bf = dcm.transposed() * velocity_air_ef;
 }
 
+void Aircraft::update_external_payload(const struct sitl_input &input)
+{
+    external_payload_mass = 0;
+
+    // update sprayer
+    if (sitl->sprayer_sim.sprayer_enable) {
```
Comment (peterbarker) [inline libraries/SITL/SIM_Aircraft.cpp]: Why not just `sitl->sprayer_sim.update()` here, passing through the sitl_input still?

File: libraries/SITL/SIM_Gripper_EPM.cpp [hunk 7]
```diff
@@ -47,10 +48,10 @@ void Gripper_EPM::update_from_demand(const Aircraft::sitl_input &input)
 
     // note that "demand" here is just an on/off switch; we only care
     // about which range it falls into
-    if (demand > 0.6) {
+    if (demand > 0.6f) {
```
[the remark this answers is not in the document]
Comment (OXINARF) [reply libraries/SITL/SIM_Gripper_EPM.cpp]: Only in NuttX and ChibiOS builds we use that flag. Also, it's likely we'll want to remove the flag someday when we want to start taking advantage of double precision FPU in new processors - so having the f is definitely a good thing to do.

**Comment by peterbarker** [comment]
Merged as part of #9063
